web/src/connectors/task-annotator-connector/task-annotator-context.tsx

```diff
@@ -407,41 +375,32 @@ export const TaskAnnotatorContextProvider: React.FC<ProviderProps> = ({
     };
 
     const deleteAnnotation = (pageNum: number, annotationId: string | number) => {
-        const pageAnnotations = allAnnotations[pageNum] ?? [];
-        const anntn: Maybe<Annotation> = pageAnnotations.find((el) => el.id === annotationId);
-        if (anntn?.labels) {
```

Comment:
is this logic moved somewhere or just deleted?

Reply:
The purpose of labels attr is to handle a case of annotation intersection. If 2 annotations are one the same place(or just ended in the same place) we show labels one above another, so when we delete one of them we still need to show another

Reply:
What do you think old code did? Could you explain it please?

As I understand:

We get the annotation by annotationId and pageNum from allAnnotations
Then we remove label from the annotation's labels.
Then we remove the annotation by annotationId and pageNum from allAnnotations
What was the point of removing label from annotation.labels if we're going to remove this annotation at all?
We just mutated object and them removed it from array.
There is no point to mutate object if we are going to delete it.

Reply:
![image](https://user-images.githubusercontent.com/69203629/233077306-9b53fdce-636d-4298-9370-ca81e41395a6.png)
in this case when you deleting label, you delete annotation but also you should remove deleted label from labels attr of other annotations.

Reply:
Was a code, which I removed, deleting labels of other annotations?
As I understand it was changing only its own labels array. I see how we get annotation by annotationId and modify it, I don't see how we get all other annotations and modify its labels.

if we need behaviour which you're describing - I suggest you to create bug/improvement to implement it as it looks like that it wasn't implemented.

Reply:
Labels are grouped by ending token and stored separately from annotations in map `tokenLabelsMap`.
So when you add or delete annotation you need to manage those labels separately, and deleting of annotation isn't enough.
I agree, that it not the clearest solution, but you need to take it into account.
I tried a case with many annotations ending on the same token on main and it works fine, but it doesn't work in your branch.